metricbeat/helper/prometheus/prometheus_test.go

```diff
@@ -412,7 +468,57 @@ func TestPrometheus(t *testing.T) {
 			},
 		},
 		{
-			msg: "Histogram metric",
+			msg:    "Summary metric using expanded layout",
+			layout: ExpandedBucketsLayout,
+			mapping: &MetricsMapping{
+				Metrics: map[string]MetricMap{
+					"summary_metric": Metric("summary.metric"),
+				},
+			},
+			expected: []common.MapStr{
+				common.MapStr{
+					"summary": common.MapStr{
+						"metric": common.MapStr{
+							"sum":   234892394.0,
+							"count": uint64(44000),
+						},
+					},
+				},
+				common.MapStr{
+					"summary": common.MapStr{
+						"metric": common.MapStr{
+							"quantile": common.MapStr{
+								"key":   "50",
+								"value": 29735.0,
+							},
+						},
+					},
+				},
+				common.MapStr{
+					"summary": common.MapStr{
+						"metric": common.MapStr{
+							"quantile": common.MapStr{
+								"key":   "90",
```

Comment:
looking at the data here, I wonder if this layout would be simpler:

```
summary.metric.quantile: 90
summary.metric.value: 47103.0
```

Reply:
But then when there are more than one quantiles, we will have more than one `summary.metric.value` hmm 
I think `summary.metric.quantile.key` and `summary.metric.quantile.value` is easier to understand 😬

Reply:
@kaiyan-sheng not sure if we are in the same page, probably my example was far from self explanatory

`httpserver.requests.duration.seconds` would be less generic example. There won't be another quantile at the same event, and there won't be any other metrics by that same name. @exekias request is 
```
httpserver.requests.duration.quantile: 90,
httpserver.requests.duration.value: 47103.0
```

vs current

```
httpserver.requests.duration.quantile.key: 90,
httpserver.requests.duration.quantile.value: 47103.0
```

reasoning I came up with the later is that when looking at the value alone, the path is telling you that it is in the context of a quantile, meanwhile you don't get that same info for `httpserver.requests.duration.value`, and could be incorrectly used for filtering if not seen at the context of the event it belongs to. I agree on the 

> ... is easier to understand

opinions?

Reply:
Thanks for your explanation. I misunderstood. So each quantile in summary will be a separate event.

Reply:
@kaiyan-sheng sorry for the delay

yes, each quantile will be an event :-(

re-thinking this, probably summaries are ok as they are now, no need to do calculations on them since they can be represented as 3 different lines.
The expansions could probably be applied only to histograms (if we get any value from expanding histograms)